libopae/api-shell.c

```diff
@@ -803,6 +803,13 @@ fpga_result fpgaPrepareBuffer(fpga_handle handle, uint64_t len, void **buf_addr,
 	opae_wrapped_handle *wrapped_handle =
 		opae_validate_wrapped_handle(handle);
 
+	// A special case: respond FPGA_OK when !buf_addr and !len
+	// as an indication that FPGA_BUF_PREALLOCATED is supported
+	// by the library.
+	if ((flags & FPGA_BUF_PREALLOCATED) && !buf_addr && !len) {
```

Comment:
Is this documented in the doxygen markup?

Reply:
It may be worth adding a \@note  to the API docs to reflect this.

Reply:
@r-rojo I added it in 'opae-c: plugins decide FPGA_BUF_PREALLOCATED'